Add nested sort functionality for DataGrid

vinothvino42 opened the pull request:
Ticket: [#688](https://github.com/EnsembleUI/ensemble/issues/688)

vusters commented on lib/layout/data_grid.dart:
```diff
@@ -474,6 +474,54 @@ class DataGridState extends WidgetState<DataGrid>
     setState(() {});
   }
 
+  void customSort(List dataList, String? sortKey, bool isAscendingOrder) {
```
extract this to a utility and add unit tests. This is not UI logic and easily testable.

vinothvino42 replied:
Done

vusters commented on lib/layout/data_grid.dart:
```diff
@@ -474,6 +474,54 @@ class DataGridState extends WidgetState<DataGrid>
+    if (sortKey == null) return;
+    final lists = JsonPath(sortKey).read(dataList).toList();
+    if (lists.length >= 2) {
+      dataList.asMap().forEach((index, value) {
```
does this even work? You loop through each item in the list then sort the entire grid? Please verify correctness throroughly with unit tests. 
Either way this logic is off. Please refactor

vinothvino42 replied:
@vusters  Something is wrong here, looks like my code is missing. Not sure, I'll look check and get back to you

kmahmood74 commented:
@vinothvino42 this is urgently needed by a customer, please look at @vusters 's comments and update quickly. 
cc @anserwaseem

kmahmood74 commented:
is it in or not in? @vusters KPN is waiting. @vinothvino42 you said your code was missing but then linked a video that shows its working? Can you please be clear here!

vinothvino42 commented:
@kmahmood74 @vusters Unfortunately my code was not pushed on Nov 2, I did it and then only I took a screen record and re-requested vu to review it and also now my local changes are missing it because I used to replace the ensemble for some reason.

It's done now. Can you review it again?

- Updated the DataGrid widget for sorting
- Moved the data grid sort logic to utils
- Wrote test cases for sorting the list with nested map objects using `key` and `order`

vusters reviewed (approved):
THanks for making the change.
